lib/collection/src/lookup.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::HashMap;
+
+use futures::Future;
+use itertools::Itertools;
+use schemars::JsonSchema;
+use segment::data_types::groups::PseudoId;
+use segment::types::{PointIdType, WithPayloadInterface, WithVector};
+use serde::{Deserialize, Serialize};
+use tokio::sync::RwLockReadGuard;
+
+use crate::collection::Collection;
+use crate::operations::consistency_params::ReadConsistency;
+use crate::operations::types::{CollectionError, CollectionResult, PointRequest, Record};
+use crate::shards::shard::ShardId;
+
+#[derive(Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(untagged)]
+pub enum Lookup {
```

Comment:
Why do we need this enum over using an Option type?

Reply:
The idea is that later this can be extended to using multiple records per lookup, as would happen if we link via arbitrary payload field. Whereas with a simple Option it would be restricted.

Reply:
Please add a comment in the code about the extensibility :)

Reply:
Done 👍